protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -1009,10 +1016,34 @@ func renderServices(services []*descriptor.Service, paths openapiPathsObject, re
 		}
 
 		for methIdx, meth := range svc.Methods {
+
+			swgReqName, ok := fullyQualifiedNameToOpenAPIName(meth.RequestType.FQMN(), reg)
+			if !ok {
+				glog.Errorf("couldn't resolve OpenAPI name for FQMN '%v'", meth.RequestType.FQMN())
+				continue
+			}
+
+			swgRspName, ok := fullyQualifiedNameToOpenAPIName(meth.ResponseType.FQMN(), reg)
+			if !ok && !skipRenderingRef(meth.ResponseType.FQMN()) {
+				glog.Errorf("couldn't resolve OpenAPI name for FQMN '%v'", meth.ResponseType.FQMN())
+				continue
+			}
+
 			if !isVisible(getMethodVisibilityOption(meth), reg) {
+				// Set request/response as invisible if not set
+				if reg.GetMessageVisibility(swgReqName) == descriptor.MessageVisibility_NOT_SET {
```

Comment:
I one is reusing a message between RPCs and only one of the RPCs is hidden, message will not be hidden. It is handled here.

`RenderServices` looks up all RPCs, and marks message as visible if one or more of RPC uses it.